presto-jdbc/src/main/java/com/facebook/presto/jdbc/PrestoConnectionConfig.java

```diff
@@ -83,6 +100,38 @@ public URI getHttpUri()
         return buildHttpUri();
     }
 
+    public Properties getConnectionProperties()
+    {
+        return connectionProperties;
+    }
+
+    public HttpClientCreator getCreator(String userAgent, JettyIoPool jettyIoPool)
```

Comment:
I would rather return here `HttpClientProperties` POJO described above. Than based on that object you could build and cache the actual `HttpClient` in `PrestoDriver`